integration-tests/deployment/ccip/test_assertions.go

```diff
@@ -14,9 +18,128 @@ import (
 	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 
 	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
 )
 
+func ConfirmGasPriceUpdatedForAll(
+	t *testing.T,
+	e deployment.Environment,
+	state CCIPOnChainState,
+	startBlocks map[uint64]*uint64,
+) {
+	var wg errgroup.Group
+	for src, srcChain := range e.Chains {
+		for dest, dstChain := range e.Chains {
+			if src == dest {
+				continue
+			}
+			srcChain := srcChain
+			dstChain := dstChain
+			wg.Go(func() error {
+				var startBlock *uint64
+				if startBlocks != nil {
+					startBlock = startBlocks[srcChain.Selector]
+				}
+				return ConfirmGasPriceUpdated(
+					t,
+					dstChain,
+					state.Chains[srcChain.Selector].FeeQuoter,
+					*startBlock,
+				)
+			})
+		}
+	}
+	require.NoError(t, wg.Wait())
+}
+
+func ConfirmGasPriceUpdated(
+	t *testing.T,
+	dest deployment.Chain,
+	srcFeeQuoter *fee_quoter.FeeQuoter,
+	startBlock uint64,
+) error {
+	it, err := srcFeeQuoter.FilterUsdPerUnitGasUpdated(&bind.FilterOpts{
+		Context: context.Background(),
```

Comment:
`tests.Context(t)` right? It will make sure that everything is terminated when the test is terminated?

Reply:
True. I followed the same we used in commit and exec checks. Will make it in a subsequent PR 👍